mesonbuild/dependencies/hdf5.py

```diff
@@ -51,7 +51,8 @@ def __init__(self, name: str, environment: 'Environment', kwargs: T.Dict[str, T.
         newinc = []  # type: T.List[str]
         for arg in self.compile_args:
             if arg.startswith('-I'):
-                stem = 'static' if kwargs.get('static', False) else 'shared'
+                static_opt =  environment.coredata.get_option(OptionKey('prefer_static'))
+                stem = 'static' if kwargs.get('static', static_opt) else 'shared'
```

Comment:
`kwargs.get('static', False)` seems sub-optimal and as a general cleanup we can probably use `self.static` here (as anywhere that inherits from ExternalDependency).